src/main/java/com/example/bookstore/controller/users/HomePageController.java

```diff
@@ -0,0 +1,28 @@
+package com.example.bookstore.controller.users;
+
+import com.example.bookstore.entity.Book;
+import com.example.bookstore.service.BooksService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+
+import java.util.List;
+
+@Controller
+public class HomePageController {
+
+    private BooksService booksService;
+
+    @Autowired
+    public HomePageController(BooksService booksService) {
+        this.booksService = booksService;
+    }
+
+    @GetMapping("/homepage")
+    public String showHomepage(Model model) {
+        List<Book> books = booksService.findAll();
+        model.addAttribute("books", books);
+        return "users/homepage";
```

Comment:
```suggestion
        return "users/home";
```
thêm chữ page làm j cho dài